--- a/neofetch
+++ b/neofetch
@@ -2084,4 +2084,15 @@
                 "ia64" | "m32r")
                     cpu="$(awk -F':' '/model/ {print $2; exit}' "$cpu_file")"
                     [[ -z "$cpu" ]] && cpu="$(awk -F':' '/family/ {printf $2; exit}' "$cpu_file")"
+                ;;
+
+                "riscv32" | "riscv64")
+                    cpu="$(awk -F':' '/^uarch[[:space:]]*:/ {print $2; exit}' "$cpu_file")"
+                    cpu="$(trim "$cpu")"
+
+                    case $cpu in
+                        "thead,c908") cpu="T-Head C908 (${kernel_machine/riscv/RISC-V})" ;;
+                        "")           cpu="${kernel_machine/riscv/RISC-V}" ;;
+                        *)             cpu="RISC-V $cpu" ;;
+                    esac
                 ;;
